Remove inline-block display from image anchor in style.scss

yogeshbhutkar opened the pull request:
Fixes: #67366 

## What?
This PR removes the display: inline-block property from the image anchor that causes the caption anchors to not flow in continuation as described in the PR.

## How?
The CSS rule causing the side-effect was removed.

## Screenshots

**Before:**
![Screenshot 2024-11-28 at 12 37 31 PM](https://github.com/user-attachments/assets/6ee500df-1020-45a3-857c-50c2462d53c4)

**After:**
![Screenshot 2024-11-28 at 12 38 50 PM](https://github.com/user-attachments/assets/a41fe4a9-cdfc-4ee1-aefc-4241062ddfc8)


### Screencast:
https://github.com/user-attachments/assets/75f34e27-bebd-431f-8feb-23f6c62954fa

yogeshbhutkar committed:
Remove inline-block display from image anchor in style.scss

t-hamano commented:
Thanks for the PR!

When deleting code, it's a good idea to use something like Git blame to find out why it's there. This code is to fix the unintended focus outline that occurs when images are linked, so we can't delete this code entirely (See https://github.com/WordPress/gutenberg/pull/62556):

![image](https://github.com/user-attachments/assets/1097865f-98d5-4893-8e13-2c79b90c3d6b)

More importantly, the HTML changes when we add left/right alignment to images in the classic theme.

No left/right alignment:

```html
<figure class="wp-block-image">
  <a href="">
    <img>
  </a>
</figure>
```

With left:right alignment:

```html
<div class="wp-block-image">
  <figure class="alignleft">
    <a href="">
      <img>
    </a>
  </figure>
</div>
```

With these things in mind, I think the following selector would be good to accurately target links to the image itself:

```scss
.wp-block-image {
	> a,
	> figure > a {
		display: inline-block;
	}
}
```

t-hamano commented:
P.S. This issue also occurs when the images are wide and full width, but those issues will be fixed in #67392.

yogeshbhutkar committed:
Refactor: Set image anchor display to inline-block in style.scss

yogeshbhutkar commented:
Thank you, @t-hamano, for reviewing the PR. I have addressed your feedback and made the necessary updates. The code has been tested and is functioning as expected. It is now ready for your review.

![Screenshot 2024-12-02 at 2 00 17 PM](https://github.com/user-attachments/assets/82369e96-7c9b-4fc6-bf25-63055afe3f94)

t-hamano reviewed (approved):
LGTM 👍
